### ISSUE TITLE
optuna.create_study.optimize execution slowly on latest versions

### ISSUE THREAD

**Issue by PerformanceDetect** [optuna/optuna#3976]
### Expected behavior

Hi, I test my program on different optuna versions.  I noticed that its execution time increased when adopting higher versions of optuna.
As shown in the following experiment results, my program performs best if we use optuna==3.0.1.

**My question is that why does such a library version perform better? Are there any issues in newer versions?**

### Environment

- Optuna version: 3.0.1, 2.10.0, 2.3.0
- Python version: 3.7.10
- OS:
Operating System: Linux Ubuntu 16.04
CPU: Intel(R) Core(TM) i9-9900K CPU
GPU: TITAN V

### Error messages, stack traces, or logs

The detailed information is as follows:
|Optuna Version|Execution Time|
|--|--|
|3.0.1|4110s|
|2.10.0|4774s|
|2.3.0|3737s|

### Steps to reproduce

1. Dataset
This dataset size>25MB, so i upload the dataset to Google Drive.
[download dataset](https://drive.google.com/file/d/1hVFrsMgnyKNFvufCLryufkc7mmc-iv7k/view?usp=sharing)
2. Code
```python
import pandas as pd
import numpy as np
import optuna
from lightgbm import LGBMRegressor

from sklearn.metrics import mean_squared_error
from sklearn.model_selection import train_test_split,KFold
from sklearn.preprocessing import LabelEncoder

train=pd.read_csv('train.csv')
test=pd.read_csv('test.csv')
cat_var=[f'cat{i}' for i in range(10)]
cont_var=[f'cont{i}' for i in range(14)]
columns=[ col for col in train.columns.tolist() if col not in ['id','target']]

for cat in cat_var:
    le = LabelEncoder()
    train[cat]=le.fit_transform(train[cat])
    test[cat]=le.transform(test[cat])
X=train[columns]
y=train.target
def objective(trial,train=train,target=y):
    X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=42)
    param={
        'num_leaves':trial.suggest_int('num_leaves',100,1000),
        'max_depth':trial.suggest_categorical('max_depth',[7,10,20,50]),
        'min_child_samples':trial.suggest_int('min_child_samples',100,500),
        'max_bin':trial.suggest_categorical('max_bin',[255,350,512,1024]),
        'learning_rate':trial.suggest_categorical('learning_rate',[0.006,0.008,0.01,0.014,0.017,0.02]),
        'colsample_bytree': trial.suggest_categorical('colsample_bytree', [0.3,0.4,0.5,0.6,0.7,0.8,0.9, 1.0]),
        'subsample': trial.suggest_categorical('subsample', [0.4,0.5,0.6,0.7,0.8,1.0]),
        'metric': 'rmse', 
        'random_state': 48,
        'n_estimators': 30000

    }
    model=LGBMRegressor(**param)
    model.fit(X_train,y_train,eval_set=[(X_test,y_test)],early_stopping_rounds=100,verbose=False)
    predictions=model.predict(X_test)
    rmse=mean_squared_error(y_test,predictions,squared=False)
    return rmse
import time
start_time = time.time()
study = optuna.create_study(direction='minimize') 
study.optimize(objective, n_trials=200)
over_time = time.time()
total_time = over_time - start_time
print("Time: %s s" % total_time)
```

**Comment by nzw0301** [comment]
Hi, could you share the lightgbm's version too?

**Comment by nzw0301** [comment]
Could you also specify the random seed of sampler when you take a benchmark https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-obtain-reproducible-optimization-results?

**Comment by not522** [comment]
I think this is not Optuna's problem. Because Optuna takes only a few seconds with `n_trials=200` and 7 parameters.
The following script runs in 4 seconds on my PC.

```python
import numpy as np

import optuna

def objective(trial):
    return np.sum([trial.suggest_float(f"x{i}", 0, 10) for i in range(10)])

def main():
    study = optuna.create_study()
    study.optimize(objective, n_trials=200)

if __name__ == "__main__":
    main()
```

There are some possible causes. For example, 

1. LightGBM's version is different.
2. LightGBM takes time with the suggested parameters.

**Comment by nzw0301** [comment]
Yeah, I agree with @not522. Some randomness or different versioned lightgbm affected the performance.
